src/components/shared/CallNodeContextMenu.js

```diff
@@ -482,10 +503,34 @@ class CallNodeContextMenuImpl extends React.PureComponent<Props> {
       ? categories[categoryIndex].name
       : '';
     const showExpandAll = selectedTab === 'calltree';
-    const canCopyURL =
-      isJS && funcTable.fileName[callNodeTable.func[callNodeIndex]] !== null;
+    const filePath = this._getFilePath();
+    const canCopyURL = isJS && filePath;
+    const fileName =
+      filePath &&
+      parseFileNameFromSymbolication(filePath)
+        .path.split('\\')
+        .pop()
+        .split('/')
+        .pop();
```

Comment:
it would be good to specify some examples so that we understand the code here. I wonder if a simple regexp such as `/[^/]+$/` could do the same thing (and possibly more easily allowing for windows-style \ path separators... would that be useful?), or using `lastIndexOf('/')` and slicing, but it's difficult to know without looking at examples.

Reply:
In Rust there's an [`rfind`](https://doc.rust-lang.org/std/primitive.str.html#method.rfind) method which lets you pass a pattern such as `s.rfind(&['/', '\\'])`, but it looks like JS does not have an equivalent of  this. So maybe a helper method which calls lastIndexOf twice would be better. Or a regex (`/([^\/\\]+)$/`), but it feels a bit wasteful.

Reply:
If the goal is indeed to find both types of path separators, then I'd go the regexp route. No need for capturing groups and no need to escape / in this case: `/[^\\/]+$/`. Simple regexps shouldn't be a performance problem, especially here as it's executed once.

Reply:
Sounds good. I said "wasteful" because I ran this regexp through the regexp debugger on https://regex101.com/ and saw how it backtracked after every slash. But since it is just run once, and on a small string, that should be fine.

Reply:
Examples of file paths:
1. `hg:hg.mozilla.org/mozilla-central:toolkit/components/startup/nsAppStartup.cpp:92bf950f388660ebc143d4a3308c0cd3f6249eb7`
2. `git:github.com/rust-lang/rust:library/std/src/sys/unix/thread.rs:53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b`

Example file names for file paths:
1. `nsAppStartup.cpp`
2. `thread.rs`

I am not sure if it's possible for `filePath` to have windows file separators. But regardless, I think matching on a regex like the one you wrote would also do the trick. Should I switch to using a regex for clarity?

Reply:
Ugh, my tab was stale and I missed out on this conversation. I'll change it to a regex for clarity when I can get to it.

Reply:
`parseFileNameFromSymbolication(fullPath).path` will strip away the colon-separated stuff.

Reply:
Oh, oops, you're already using that.

Reply:
> Sounds good. I said "wasteful" because I ran this regexp through the Regex Debugger on https://regex101.com/ and saw how it backtracked after every slash. But since it is just run once, and on a small string, that should be fine.

ahah I looked at this website but couldn't find the debugger because I switched to Javascript mode first... There are ways to optimize the regexp (negative look behind comes to mind but Safari doesn't support it, so we could emulate it with a capture group), but probably not worth it here.